builders/tools/multisig-safe.md

```diff
@@ -0,0 +1,352 @@
+        text = _text;
+    }
+}
+```
+
+这是一个带有单个函数`setTextData`的简单合约，接受一个字符串并使用它来设置`text`变量。
+
+您将需要合约地址和ABI，请确保您已将它们复制或保存到其他地方，以备不时之需。
+
+## 创建一个Safe钱包
+
+要创建一个Safe钱包，导航至[Moonbeam Safe](https://multisig.moonbeam.network/moonbase)。本教程将引导您在Moonbase Alpha上创建一个Safe钱包，您也可以修改配置[在Moonriver上创建一个Safe钱包](https://multisig.moonbeam.network/moonriver)。您可点击页面右上角的网络下拉菜单切换网络。
+
+### 连接MetaMask
+
+进入[Moonbase Alpha](https://multisig.moonbeam.network/moonbase/)页面后，开始创建Safe钱包之前先连接您的钱包：
+
+ 1. 点击**Connect Wallet**
+
+  2. 选择钱包连接至Moonbeam Safe。本示例中您可以使用MetaMask。如果MetaMask没有出现在选项列表中，点击**Show More**并选择**MetaMask**
+
+![Connect Wallet to Moonbeam Safe](/images/builders/tools/multisig-safe/safe-1.png)
+
+如果您还没有登录MetaMask，系统将会提示您登录。随后，将引导您添加和连接您的帐户，以及添加和切换至Moonbase Alpha网络：
+
+ 1. 选择账户并连接至Safe钱包。您将需要选择至少3个所有者账户中的其中两个，然后点击**Next**。本示例中，将选择Alice、Bob和Charlie账户。
+
+ 2. 点击**Connect**连接至已选账户
+
+ 3. 如果您还没有连接至Moonbase Alpha，或者您还没有在您的MetaMask添加Moonbase Alpha网络，点击**Approve**将Moonbase Alpha添加为自定义网络
+
+  4. 点击**Switch Network**将网络切换至Moonbase Alpha
+
+![Connect MetaMask to Moonbase Alpha](/images/builders/tools/multisig-safe/safe-2.png)
+
+现在，您可以在右上方确认您的MetaMask账户已连接至Moonbase Alpha网络。如果您使用的是开发账户，您应该可以看到Alice的账户地址。如果不是，请再次确认您的MetaMask并切换到Alice账户。
+
+### 创建新的Safe钱包
+
+点击**Create new Safe**，在Moonbase Alpha上创建一个新的Safe钱包。页面将出现一个向导，引导您完成创建新的Safe钱包。完成这些步骤并创建新的Safe钱包，即表示您同意使用条款和隐私政策。因此，在开始之前，请仔细查看这些内容。
+
+![Create Safe](/images/builders/tools/multisig-safe/safe-3.png)
+
+您将需要给您的Safe钱包设置名称：
+
+ 1. 输入您的新Safe钱包名称，您可以使用`moonbeam-tutorial`
+
+  2. 点击**Start**
+
+![Submit Safe Name](/images/builders/tools/multisig-safe/safe-4.png)
+
+接下来是所有者和签名确认部分。在这个部分中，您将添加Safe钱包的所有者并设置阈值。阈值决定了在交易执行之前需要多少位所有者来签名确认。
+
+创建Safe钱包的时候可以使用许多不同的设置。Safe钱包可以有1个或多个所有者以及不同的阈值级别。请注意，不建议创建只有1个所有者的Safe钱包，因为它会产生单点故障的可能性。
+
+在本教程中，您将创建一个具有3个所有者并要求阈值为2的多重签名设置，因此至少需要3个所有者中的2人使用私钥签署才能通过Safe钱包执行交易。
+
+您的帐户将自动预先填写为第一个所有者，但如果您想使用不同的帐户，则可以更改此设置。在本示例中，Alice的帐户已预先填写。除了Alice之外，您还可以添加Bob和Charlie作为所有者：
+
+ 1. 点击**Add another owner**
+
+ 2. 输入**Bob**作为第二个所有者，以及他的地址：`0x3Cd0A705a2DC65e5b1E1205896BaA2be8A07c6e0`
+
+ 3. 输入**Charlie**作为第三个所有者，以及他的地址：`0x798d4Ba9baf0064Ec19eB4F0a1a45785ae9D6DFc`
+
+ 4. 设置3个所有者的确认阈值为**2**
+
+  5. 点击**Review**进入向导的最后步骤
+
+![Enter Safe Owners](/images/builders/tools/multisig-safe/safe-5.png)
+
+最后，您可以查看所有Safe钱包和所有者的详细信息，确认信息无误后：
+
+ 1. 点击**Submit**创建新的Safe钱包。这将在Moonbase Alpha上产生约少于0.001 DEV的手续费。MetaMask将会跳出弹窗提示您确认签署交易。
+
+  2. 点击**Confirm**发送交易并创建Safe钱包
+
+![Send Transaction to Create Multisig Safe](/images/builders/tools/multisig-safe/safe-6.png)
+
+处理交易和创建Safe钱包可能需要几分钟时间。创建成功后，您将会看到一条消息，提示“Your Safe was created successfully”。然后，您可以点击**Get Started**加载您的Safe钱包并开始交互。
+
+![Safe Created Successfully](/images/builders/tools/multisig-safe/safe-7.png)
+
+## 配置Safe钱包
+
+您可以随时管理您的Safe钱包并在创建时更改一些参数设置。为此，您可以点击左侧菜单上的**Settings**选项。
+
+![Modify Safe Settings](/images/builders/tools/multisig-safe/safe-8.png)
+
+随后，您将会看到以下选项：
+
+ - **Safe Details** —— 允许您更改Safe钱包名称。这是一个无需链上交互的本地操作
```

Comment:
```suggestion
 - **Safe Details** — 允许您更改Safe钱包名称。这是一个无需链上交互的本地操作
```

Reply:
these dashes are displaying oddly on the page, all of the items in this list need to be modified so there is only the one long dash